cpp/python/include/ucxx/python/python_future_task.h

```diff
@@ -0,0 +1,213 @@
+    } catch (const std::out_of_range& e) {
+      setPythonException(PyExc_IndexError, e.what());
+    } catch (const std::overflow_error& e) {
+      setPythonException(PyExc_OverflowError, e.what());
+    } catch (const std::range_error& e) {
+      setPythonException(PyExc_ArithmeticError, e.what());
+    } catch (const std::underflow_error& e) {
+      setPythonException(PyExc_ArithmeticError, e.what());
+    } catch (const std::exception& e) {
+      setPythonException(PyExc_RuntimeError, e.what());
+    } catch (...) {
+      setPythonException(PyExc_RuntimeError, "Unknown exception");
+    }
+  }
+
+ public:
+  /**
+   * @brief Construct a Python future backed by C++ `std::packaged_task`.
+   *
+   * Construct a future object that receives a user-defined C++ `std::packaged_task` which
+   * runs asynchronously using an internal `std::async` that ultimately notifies a Python
+   * future that can be awaited in Python code.
+   *
+   * @param[in] task the user-defined C++ task.
+   * @param[in] pythonConvert C-Python function to convert a C object into a `PyObject*`
+   *                          representing the result of the task.
+   * @param[in] asyncioEventLoop pointer to a valid Python object containing the event loop
+   *                             that the application is using, to which the Python future
+   *                             will belong to.
+   * @param[in] launchPolicy launch policy for the async C++ task.
+   */
+  explicit PythonFutureTask(std::packaged_task<ReturnType(TaskArgs...)> task,
+                            std::function<PyObject*(ReturnType)> pythonConvert,
+                            PyObject* asyncioEventLoop,
+                            std::launch launchPolicy = std::launch::async)
+    : _task{std::move(task)},
+      _pythonConvert(pythonConvert),
+      _asyncioEventLoop(asyncioEventLoop),
+      _handle{ucxx::python::create_python_future_with_event_loop(asyncioEventLoop)}
+  {
+    _future = std::async(launchPolicy, [this]() {
+      std::future<ReturnType> result = this->_task.get_future();
+      this->_task();
+      try {
+        const ReturnType r = result.get();
+        this->setResult(r);
+        return r;
+      } catch (std::exception& e) {
+        this->setException(e);
+      }
+    });
+  }
+  PythonFutureTask(const PythonFutureTask&)            = delete;
+  PythonFutureTask& operator=(PythonFutureTask const&) = delete;
+  PythonFutureTask(PythonFutureTask&& o)               = delete;
+  PythonFutureTask& operator=(PythonFutureTask&& o)    = delete;
+
+  /**
+   * @brief Python future destructor.
+   *
+   * Decrement the reference count of the underlying Python future.
+   */
+  ~PythonFutureTask() { Py_XDECREF(_handle); }
+
+  /**
+   * @brief Get the C++ future.
+   *
+   * Get the underlying C++ future that can be awaited and have its result read.
+   *
+   * @returns The underlying C++ future.
+   */
+  std::future<ReturnType>& getFuture() { return _future; }
```

Comment:
Since the other _getters_ are guarded against invalid object, we should do it here as well? 
```suggestion
  std::future<ReturnType>& getFuture() { 
    if (_handle == nullptr) throw std::runtime_error("Invalid object or already released");
    
    return _future; 
  }
```